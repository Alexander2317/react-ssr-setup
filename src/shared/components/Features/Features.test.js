import React from 'react';
import renderer from 'react-test-renderer';
import Features from '.';
import { shallow } from 'enzyme';

jest.mock('react-i18next', () => ({
    withNamespaces: () => (Component) => {
        Component.defaultProps = { ...Component.defaultProps, t: (key) => `[translated] ${key}` };
        return Component;
    },
}));

describe('Features component', () => {
    it('tests if the component is rendered properly', () => {
        expect(renderer.create(<Features />).toJSON()).toMatchSnapshot();
    });

    it('tests the component contains a list of features', () => {
        expect(shallow(<Features />).find('ul').length).toEqual(1);
    });
});
