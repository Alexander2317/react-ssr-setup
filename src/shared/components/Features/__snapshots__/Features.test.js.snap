// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features component tests if the component is rendered properly 1`] = `
Array [
  <h2>
    [translated] features
  </h2>,
  <ul>
    <li>
      Webpack 4
    </li>
    <li>
      Babel 7
    </li>
    <li>
      ESLint 5
    </li>
    <li>
      Flow Type
    </li>
    <li>
      React 16.6
    </li>
    <li>
      React Router 4
    </li>
    <li>
      Redux (+ Thunk)
    </li>
    <li>
      Reselect
    </li>
    <li>
      React Helmet
    </li>
    <li>
      Express Webserver + Server Side Prerendering
    </li>
    <li>
      [translated] i18n-support
    </li>
    <li>
      Jest 24
    </li>
    <li>
      CSS Modules
    </li>
    <li>
      PostCSS
    </li>
    <li>
      Prettier (incl. precommit-hook via lint-staged + husky)
    </li>
    <li>
      HMR (buggy, see Readme)
    </li>
  </ul>,
]
`;
